pkg/auth/providers/activedirectory/activedirectory_provider.go

```diff
@@ -229,9 +229,13 @@ func (p *adProvider) getDNAndScopeFromPrincipalID(principalID string) (string, s
 	return externalID, scope, nil
 }
 
-func (p *adProvider) GetUserExtraAttributes(token *v3.Token) map[string][]string {
+func (p *adProvider) GetUserExtraAttributes(userPrincipal v3.Principal) map[string][]string {
 	extras := make(map[string][]string)
-	extras["principalid"] = []string{token.UserPrincipal.Name}
-	extras["username"] = []string{token.UserPrincipal.LoginName}
+	if userPrincipal.Name != "" {
+		extras["principalid"] = []string{userPrincipal.Name}
```

Comment:
Somewhat of a nitpick, but it could be a good idea to introduce consts for repeating string literals like `"principalid"` and `"username"`.

Reply:
sure